---
title: "\n Analysis Performed on 2021-07-22"
date: "2021-07-22"
output: 
  html_document
---
```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(DT)
library(ggplot2)
library(ggtree)
library(knitr)
library(plotly)
library(RColorBrewer)
library(reshape2)
library(treeio)
library(kableExtra)
library(here)
library(magick)
#webshot::install_phantomjs()
#knitr::opts_chunk$set(out.width="1200px", dpi=120, fig.align = "center")
knitr::opts_chunk$set(dpi=120, fig.align = "center")
options(digits = 3) 
```

TORMES version 1.3.0

<br>

### Quick link to sections {.sidebar #index}
* #### [Samples description](#description)
* #### [Assembly statistics](#assembly)
* #### [Taxonomic identification](#taxonomy)
* #### [Multi-Locus Sequence Typing (MLST)](#mlst)
* #### [Pangenome analysis](#pangenome)
* #### [Antibiotic resistance](#resistance)
* #### [Virulence genes](#virulence)
* #### [Citation](#cite)

<br>

## Samples description {#description}
```{r, echo=FALSE}
setwd("/Users/d1794974/Documents/PhD/Wood_ants/bacterial_genomes/tormes")
data=read.table("metadata.txt", header = T, sep = "\t", dec = ".", check.names = FALSE)
if (ncol(data)<4){
	metadata=as.data.frame(data[,1])
	print ("No more data regarding samples was provided")
} else {
	data=data[-2]
	metadata=data[-2]
}
colnames(metadata)[1]="Samples"
datatable(metadata, rownames=F, fillContainer = F, filter = 'top',
          class='cell-border stripe', style = 'default',
  options = list(
  pageLength = 20, autoWidth = TRUE,
  columnDefs = list(list(className='dt-center', targets="_all"))
  )
)
```

<br>

[Back to index](#index)

## Assembly statistics {#assembly}
<br>

Field | Description
----- | ---------------------------------------------------------------------------------------
**Sample** | Name of the sample
**Reads** | Total number of reads after quality filtering
**AvgReadLen** | Average read length after quality filtering
**Contigs** | Number of contigs of the draft genome (>200bp)
**GenomeLength** | Length (bp) of the draft genome
**LargestContig** | Length (bp) of the largest contig in the genome
**N50** | "Length of the smallest contig in the set that contains the fewest (largest) contigs whose combined length represents at least 50% of the assembly" ([Miller *et al*., 2010](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2874646/))
**GC** | GC content (%) of the draft genome
**Depth** | Number of times each nucleotide position in the draft genome has a read that align to that position

<br>

\* *When already assembled genomes are included for TORMES analysis, "NA" will appear for "Reads", "AvgReadLen" and "Depth" fields.*

<br>

```{r, echo=FALSE}
seqdata=read.table("sequencing_assembly_report.txt", header = T, sep = "\t", dec = ".")
colnames(seqdata)=c("Sample", "Reads", "AvgReadLength", "Contigs", "GenomeLength", "LargestContig", "N50", "GC", "Depth")

seqdata <- transmute(seqdata, Sample, Reads, AvgReadLength, Contigs, GenomeLengthMb = GenomeLength / 1000000, LargestContigMb = LargestContig / 1000000, N50, GC, Depth)

colnames(seqdata)=c("Sample", "Reads", "Average Read Length (bp)", "Contigs", "Genome Length (Mb)", "Largest Contig (Mb)", "N50", "GC (%)", "Depth")
datatable(seqdata, rownames=F, fillContainer = F, filter = 'top',
          class='cell-border stripe', style = 'default',
  options = list(
  pageLength = 20, autoWidth = TRUE,
  columnDefs = list(list(className='dt-center', targets="_all"))
  
  )
)

seqdata <- as_tibble(seqdata)

kable(seqdata, "html") %>%
  kable_styling() %>%
  save_kable("summary_stats.pdf") 


```

<br>

#### Select the parameter {.tabset .tabset-fade .tabset-pills}
##### Number of contigs
```{r, echo=FALSE}
x=seqdata$Sample
y=seqdata$Contigs
a <- list(title = "Number of contigs", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average number of contigs per sample = **`r mean(seqdata$Contigs)`**

##### Genome length
```{r, echo=FALSE}
x=seqdata$Sample
y=seqdata$GenomeLength
a <- list(title = "Genome length", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average genome length per sample = **`r mean(seqdata$GenomeLength)` bp**

##### Largest contig length
```{r, echo=FALSE}
x=seqdata$Sample
y=seqdata$LargestContig
a <- list(title = "Largest contig length", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average largest contig length per sample = **`r mean(seqdata$LargestContig)` bp**

##### N50
```{r, echo=FALSE}
x=seqdata$Sample
y=seqdata$N50
a <- list(title = "N50", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average N50 per sample = **`r mean(seqdata$N50)` bp**

##### GC content
```{r, echo=FALSE}
options("scipen"=0, "digits"=3)
x=seqdata$Sample
y=seqdata$GC
a <- list(title = "GC content (%)", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average GC content per sample = **`r mean(seqdata$GC)`%**

##### Number of reads
```{r, echo=FALSE}
options("scipen"=100, "digits"=0)
x=seqdata$Sample
y=seqdata$Reads
a <- list(title = "Number of reads", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average number of reads per sample = **`r mean(seqdata$Reads, na.rm=TRUE)`**

##### Average read length
```{r, echo=FALSE}
x=seqdata$Sample
y=seqdata$AvgReadLen
a <- list(title = "Average read length", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average read length per sample = **`r mean(seqdata$AvgReadLen, na.rm=TRUE)` bp**

##### Sequencing depth
```{r, echo=FALSE}
seqdata$Depth=gsub('X','', seqdata$Depth)
seqdata$Depth=as.numeric(seqdata$Depth)
x=seqdata$Sample
y=seqdata$Depth
a <- list(title = "Sequencing depth (X)", showticklabels = TRUE, exponentformat = "E")
plot_ly(x=x, y=y, type = "bar") %>%
layout(yaxis = a)
```
Average sequencing depth = **`r mean(seqdata$Depth, na.rm=TRUE)`X**



#

<br>

[Back to index](#index)

## Taxonomic identification {#taxonomy}
<br>
Taxonomic identification was performed by using Kraken2 ([Wood *et al*., 2019](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0)). Further details can be found in the [Kraken2 webpage](https://ccb.jhu.edu/software/kraken2/index.shtml).
Additionally, 16S rRNA genes were extracted from each genome by using [Barrnap](https://github.com/tseemann/barrnap) and used for taxonomic classification by using the RDP Classifier ([Wang *et al*., 2007](https://aem.asm.org/content/73/16/5261)) at a confidence level of 0.8. Further details can be found in the [RDPTools webpage](https://github.com/rdpstaff/classifier).

<br>

### Taxonomic identification by using Kraken2

The number between brackets refers to the percentage of reads (when starting TORMES from raw reads) or contigs (when starting TORMES from already assembled genomes) from each sample covered by the clade rooted at this taxon.

<br>

```{r, echo=FALSE}
kraken2=read.table("taxonomic-identification-kraken2.txt", header = T, sep = "\t", dec = ".")
colnames(kraken2)=c("Sample", "Family", "Genus", "Species")
kable(kraken2, results='asis', align = "c")

print(xtable(kraken2, type = "latex"), file= "kraken2_results.tex", include.rownames=FALSE)
```

<br>

### Taxonomic identification by using RDP Classifier

<br>

Field | Description
----- | ---------------------------------------------------------------------------------------
**Sample** | Name of the sample where the 16S rRNA gene was found. Note that the same sample might harbor more than one 16S rRNA gene copy.
**Contig** | Name of the contig where the 16S rRNA gene was found.
**Position** | Position within the contig where the 16S rRNA gene was found (begin-end). Note that the 16S rRNA genes identified for one sample might not harbor the same length (fragmented, *etc.*).
**Strand** | Strand of the 16S rRNA gene defined as forward (+) or reverse (-).
**Order** | Taxonomic order that the 16S rRNA gene was assigned over a confidence level of 0.8.
**Family** | Taxonomic family that the 16S rRNA gene was assigned over a confidence level of 0.8.
**Genus** | Taxonomic genus that the 16S rRNA gene was assigned over a confidence level of 0.8.

<br>

```{r, echo=FALSE, error=TRUE}
rdp=read.table("16S-rRNA.RDP.txt", header = T, sep = "\t", dec = ".")
colnames(rdp)=c("Sample", "Contig", "Position", "Strand", "Order", "Family", "Genus")
kable(rdp, results='asis', align = "c")

print(xtable(rdp, type = "latex"), file= "rdp_results.tex", include.rownames=FALSE)

```

<br>

[Back to index](#index)

## Multi-Locus Sequence Typing (MLST) {#mlst}

Further details can be found in [mlst web page](https://github.com/tseemann/mlst).

Symbol | Meaning
------ | ---------------------------------------------------------------------------------------
~ | novel full length allele similar to match
? | partial match to known allele
- | allele missing

<!-- <br> -->

<!-- ```{r, echo=FALSE} -->
<!-- mlst=read.table("mlst.tab", header = F, sep = "\t", dec = ".", row.names=1, fill=TRUE) -->
<!-- mlst=as.data.frame(mlst) -->
<!-- mlst=mlst[order(row.names(mlst)),] -->
<!-- colnames(mlst)[1]="Scheme" -->
<!-- colnames(mlst)[2]="ST" -->
<!-- if (ncol(mlst)>=3) { -->
<!-- for (i in 3:ncol(mlst)){ -->
<!-- 	colnames(mlst)[i]=i-2 -->
<!-- } } -->
<!-- kable(mlst, results='asis', align = "c") -->
<!-- ``` -->

<!-- <br> -->

[Back to index](#index)

## Pangenome analysis {#pangenome}
Pangenome analysis was not performed.

<br>

[Back to index](#index)

## Antimicrobial resistance {#resistance}
The search of antimicrobial resistance genes was performed by screening the genome against Resfinder ([E. Zankari *et al*., 2012](https://www.ncbi.nlm.nih.gov/pubmed/22782487)), CARD ([A.G. McArthur *et al*., 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3697360/)) and ARG-ANNOT ([S.K. Gupta *et al*., 2014](https://www.ncbi.nlm.nih.gov/pubmed/24145532)) databases by using [Abricate](https://github.com/tseemann/abricate).
Any hit with coverage and/or identity below 90% was removed. If you would like to inspect hits with coverage and/or identity below 90%, please go to the corresponing TORMES output directory.

<br>

Field | Description
----- | -------------------------------------------------------------------------
**Contig** | Contig name where the gene was found
**Start** | Start position of the gene within the contig
**End** | End position of the gene within the contig
**Gene** | Antimicrobial resistance gene
**Gaps** | Gaps in subject (genome) and query (antimicrobial resistance gene)
**Coverage** | Proportion (%) of the gene covered
**Identity** | Proportion (%) of exact nucleotide matches
**Accession** | Accesion number of the antimicrobial resistance gene
**Product** | Antimicrobial resistance gene product
**Resistance** | Antimicrobial that the gene confers resistance to

<br>

<!-- #### Select the samples and the database {.tabset .tabset-fade .tabset-pills} -->
<!-- ##### Isolate_1 {.tabset .tabset-fade .tabset-pills} -->
<!-- ###### Resfinder -->
<!-- ```{r, echo=FALSE} -->
<!-- ABresfinder=read.table("Isolate_1_resfinder.tab", header = T, sep = "\t", dec = ".", quote = "") -->
<!-- ABresfinder=as.data.frame(ABresfinder) -->
<!-- Resfdata=cbind(ABresfinder[,2:4], ABresfinder[,6], ABresfinder[,9:11], ABresfinder[,13:15]) -->
<!-- colnames(Resfdata)=c("Contig", "Start", "End", "Gene", "Gaps", "Coverage", "Identity", "Accession", "Product", "Resistance") -->
<!-- ABresfinder = Resfdata[Resfdata$Identity>90,] -->
<!-- ABresfinder = ABresfinder[ABresfinder$Coverage>90,] -->
<!-- ABresfinder = ABresfinder[order(ABresfinder$Gene),] -->
<!-- row.names(ABresfinder)=NULL -->
<!-- if (nrow(ABresfinder)==0){ -->
<!--   print ("No antimicrobial resitance genes from Resfinder database were found") -->
<!-- } else { -->
<!-- kable(ABresfinder, results='asis', align="c") -->
<!-- } -->
<!-- ``` -->

<!-- ###### CARD -->
<!-- ```{r, echo=FALSE} -->
<!-- ABcard=read.table("Isolate_1_card.tab", header = T, sep = "\t", dec = ".", quote = "") -->
<!-- ABcard=as.data.frame(ABcard) -->
<!-- carddata=cbind(ABcard[,2:4], ABcard[,6], ABcard[,9:11], ABcard[,13:15]) -->
<!-- colnames(carddata)=c("Contig", "Start", "End", "Gene", "Gaps", "Coverage", "Identity", "Accession", "Product", "Resistance") -->
<!-- ABcard = carddata[carddata$Identity>90,] -->
<!-- ABcard = ABcard[ABcard$Coverage>90,] -->
<!-- ABcard = ABcard[order(ABcard$Gene),] -->
<!-- row.names(ABcard)=NULL -->
<!-- if (nrow(ABcard)==0){ -->
<!--   print ("No antimicrobial resitance genes from CARD database were found") -->
<!-- } else { -->
<!-- kable(ABcard, results='asis', align="c") -->
<!-- } -->
<!-- ``` -->

<!-- ###### ARG-ANNOT -->
<!-- ```{r, echo=FALSE} -->
<!-- ABargannot=read.table("Isolate_1_argannot.tab", header = T, sep = "\t", dec = ".", quote = "") -->
<!-- ABargannot=as.data.frame(ABargannot) -->
<!-- argannotdata=cbind(ABargannot[,2:4], ABargannot[,6], ABargannot[,9:11], ABargannot[,13:15]) -->
<!-- colnames(argannotdata)=c("Contig", "Start", "End", "Gene", "Gaps", "Coverage", "Identity", "Accession", "Product", "Resistance") -->
<!-- ABargannot = argannotdata[argannotdata$Identity>90,] -->
<!-- ABargannot = ABargannot[ABargannot$Coverage>90,] -->
<!-- ABargannot = ABargannot[order(ABargannot$Gene),] -->
<!-- row.names(ABargannot)=NULL -->
<!-- if (nrow(ABargannot)==0){ -->
<!--   print ("No antimicrobial resitance genes from ARG-ANNOT database were found") -->
<!-- } else { -->
<!-- kable(ABargannot, results='asis', align="c") -->
<!-- } -->
<!-- ``` -->


<!-- # -->

<!-- <br> -->

#### Summary figures
Presence/Absence of antimicrobial resistance genes (coverage and identity > 90%) on each sample.

<br>

<!-- ##### Select the database: {.tabset .tabset-fade .tabset-pills} -->
<!-- ###### Resfinder -->
<!-- ```{r, echo=FALSE, message=FALSE, error=TRUE} -->
<!-- ResfinderSummary=read.table("resfinder_summary.tab", header = T, dec = ".", sep = "\t", check.names = FALSE) -->
<!-- ResfinderSummary=as.data.frame(ResfinderSummary) -->
<!-- if (sum(ResfinderSummary$NUM_FOUND)==0) { -->
<!--   print ("No antimicrobial resitance genes from ResFinder database were found") -->
<!-- } else { -->
<!-- ResfinderSummary=ResfinderSummary[,-2] -->
<!-- Resfinderdatos=as.matrix(ResfinderSummary[,-1]) -->
<!-- x=colnames(ResfinderSummary)[-1] -->
<!-- y=ResfinderSummary$FILE -->
<!-- plot_ly(x=x, y=y, z=Resfinderdatos, type = "heatmap", colors=colorRamp(c("gainsboro", "olivedrab", "forestgreen")), showscale=F) %>% layout(xaxis=list(type="category"), yaxis = list(autorange="reversed")) -->
<!-- } -->
<!-- ``` -->

<!-- ###### CARD -->
<!-- ```{r, echo=FALSE, message=FALSE, error=TRUE} -->
<!-- CARDSummary=read.table("card_summary.tab", header = T, dec = ".", sep = "\t", check.names = FALSE) -->
<!-- CARDSummary=as.data.frame(CARDSummary) -->
<!-- if (sum(CARDSummary$NUM_FOUND)==0) { -->
<!--   print ("No antimicrobial resitance genes from CARD database were found") -->
<!-- } else { -->
<!-- CARDSummary=CARDSummary[,-2] -->
<!-- CARDdatos=as.matrix(CARDSummary[,-1]) -->
<!-- x=colnames(CARDSummary)[-1] -->
<!-- y=CARDSummary$FILE -->
<!-- plot_ly(x=x, y=y, z=CARDdatos, type = "heatmap", colors=colorRamp(c("gainsboro", "firebrick1", "firebrick3")), showscale=F) %>% layout(xaxis=list(type="category"), yaxis = list(autorange="reversed")) -->
<!-- } -->
<!-- ``` -->

<!-- ###### ARG-ANNOT -->
<!-- ```{r, echo=FALSE, message=FALSE, error=TRUE} -->
<!-- ARGSummary=read.table("argannot_summary.tab", header = T, dec = ".", sep = "\t", check.names = FALSE) -->
<!-- ARGSummary=as.data.frame(ARGSummary) -->
<!-- if (sum(ARGSummary$NUM_FOUND)==0) { -->
<!--   print ("No antimicrobial resitance genes from ARG-ANNOT database were found") -->
<!-- } else { -->
<!-- ARGSummary=ARGSummary[,-2] -->
<!-- ARGdatos=as.matrix(ARGSummary[,-1]) -->
<!-- x=colnames(ARGSummary)[-1] -->
<!-- y=ARGSummary$FILE -->
<!-- plot_ly(x=x, y=y, z=ARGdatos, type = "heatmap", colors=colorRamp(c("gainsboro", "dodgerblue1", "dodgerblue3")), showscale=F) %>% layout(xaxis=list(type="category"), yaxis = list(autorange="reversed")) -->
<!-- } -->
<!-- ``` -->

<!-- # -->


<!-- <br> -->

<!-- [Back to index](#index) -->

<!-- ## Virulence genes {#virulence} -->
<!-- The search of virulence genes was performed by screening the genome against Virulence Factors Data Base (VFDB, [L. Chen *et al*., 2005](https://www.ncbi.nlm.nih.gov/pubmed/15608208)) by using [Abricate](https://github.com/tseemann/abricate). Any hit with coverage and/or identity below 90% was removed. If you would like to inspect hits with coverage and/or identity below 90%, please go to the corresponing TORMES output directory. -->

<!-- <br> -->

<!-- Field | Description -->
<!-- ----- | ------------------------------------------------------------------------- -->
<!-- **Contig** | Contig name where the virulence gene was found -->
<!-- **Start** | Start position of the virulence gene within the contig -->
<!-- **End** | End position of the virulence gene within the contig -->
<!-- **Gene** | Virulence gene -->
<!-- **Gaps** | Gaps in subject (genome) and query (virulence gene) -->
<!-- **Coverage** | Proportion (%) of the gene covered -->
<!-- **Identity** | Proportion (%) of exact nucleotide matches -->
<!-- **Accession** | Accesion number of the virulence gene -->
<!-- **Product** | Virulence gene product -->

<!-- <br> -->

<!-- #### Select the sample {.tabset .tabset-fade .tabset-pills} -->
<!-- ##### Isolate_1 -->
<!-- ```{r, echo=FALSE, message=FALSE, error=TRUE} -->
<!-- vfdb=read.table("Isolate_1_vfdb.tab", header = T, sep = "\t", dec = ".", quote = "") -->
<!-- vfdb=as.data.frame(vfdb) -->
<!-- vfdbdata=cbind(vfdb[,2:4], vfdb[,6], vfdb[,9:11], vfdb[,13:14]) -->
<!-- colnames(vfdbdata)=c("Contig", "Start", "End", "Gene", "Gaps", "Coverage", "Identity", "Accession", "Product") -->
<!-- vfdb = vfdbdata[vfdbdata$Identity>90,] -->
<!-- vfdb = vfdb[vfdb$Coverage>90,] -->
<!-- vfdb = vfdb[order(vfdb$Gene),] -->
<!-- row.names(vfdb)=NULL -->
<!-- if (nrow(vfdb)==0){ -->
<!--   print ("No virulence genes from VFDB database were found") -->
<!-- } else { -->
<!-- kable(vfdb, results='asis', align="c") -->
<!-- } -->
<!-- ``` -->


<!-- # -->

<!-- <br> -->

<!-- [Back to index](#index) -->

## Citation  {#cite}
If using **TORMES**, please cite it as:

<br>

<center>

Narciso Martin Quijada, David Rodriguez-Lazaro, Jose Maria Eiros, Marta Hernandez (2019).
TORMES: an automated pipeline for whole bacterial genome analysis.
*Bioinformatics*, 35(21), 4207â€“4212, https://doi.org/10.1093/bioinformatics/btz220.

</center>

<br>

TORMES is a pipeline that relies in many different software and databases that should be cited when using TORMES.
The following software and databases were used for this analysis:

* GNU Parallel, [O. Tange, 2018](https://doi.org/10.5281/zenodo.1146014)
* Prinseq, [R. Schmieder and R. Edwards, 2011](https://www.ncbi.nlm.nih.gov/pubmed/21278185)
* SPAdes, [A. Bankevich *et al*., 2012](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/)
* QUAST, [A. Gurevich *et al*., 2013](https://www.ncbi.nlm.nih.gov/pubmed/23422339)
* Barrnap, [T. Seemann](https://github.com/tseemann/barrnap)
* Kraken2, [Wood *et al*., 2019](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0)
* RDP Classifier, [Q. Wang *et al*., 2007](https://aem.asm.org/content/73/16/5261)
* mlst, [T. Seemann](https://github.com/tseemann/mlst)
* ABRicate, [T. Seemann](https://github.com/tseemann/abricate)
* ResFinder database, [E. Zankari *et al*., 2012](https://www.ncbi.nlm.nih.gov/pubmed/22782487)
* CARD database, [A.G. McArthur *et al*., 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3697360/)
* ARG-ANNOT database, [S.K. Gupta *et al*., 2014](https://www.ncbi.nlm.nih.gov/pubmed/24145532)
* VFDB database, [L. Chen *et al*., 2005](https://www.ncbi.nlm.nih.gov/pubmed/15608208)
* Prodigal, [Hyatt *et al*., 2010](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2848648/)
* Prokka, [T. Seemann, 2014](https://www.ncbi.nlm.nih.gov/pubmed/24642063)
* R, [R Development Core Team, 2008](https://www.r-project.org/)
  + ggplot2, [H. Wickham, 2009](https://cran.r-project.org/web/packages/ggplot2/index.html)
  + ggtree, [G. Yu *et al*., 2016](https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12628)
  + knitr, [Y. Xie, 2015](https://cran.r-project.org/web/packages/knitr/index.html)
  + plotly, [C. Sievert *et al*., 2017](https://cran.r-project.org/web/packages/plotly/index.html)
  + RColorBrewer, [E. Neuwirth and R.C. Brewer, 2014](https://cran.r-project.org/web/packages/RColorBrewer/index.html)
  + reshape2, [H. Wickham, 2007](https://cran.r-project.org/web/packages/reshape2/index.html)
  + rmarkdown, [J.J. Allaire, 2015](https://cran.r-project.org/web/packages/rmarkdown/index.html)
  + treeio, [L-G. Wang *et al*., 2019](https://academic.oup.com/mbe/article-abstract/37/2/599/5601621?redirectedFrom=fulltext)

[Back to index](#index)

